### PR TITLE
Remove the deprecated load_sample_bathymetry function (deprecated since v0.6.0)

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Related to #2302

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/samples.py [hunk 1]
```diff
@@ -179,35 +179,17 @@ def load_ocean_ridge_points(**kwargs):
...
+def _load_sample_bathymetry():
     """
-    (Deprecated) Load a table of ship observations of bathymetry off Baja
-    California as a pandas.DataFrame.
-
-    .. warning:: Deprecated since v0.6.0. This function has been replaced with
-       ``load_sample_data(name="bathymetry")`` and will be removed in
-       v0.9.0.
-
-    This is the ``@tut_ship.xyz`` dataset used in the GMT tutorials.
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:191]: Maybe we should keep this background information of the data?

File: pygmt/datasets/samples.py [hunk 2]
```diff
@@ -179,35 +179,17 @@ def load_ocean_ridge_points(**kwargs):
...
-
-    The data are downloaded to a cache directory (usually ``~/.gmt/cache``) the
-    first time you invoke this function. Afterwards, it will load the data from
-    the cache. So you'll need an internet connection the first time around.
+    Load a table of ship observations of bathymetry off Baja California as a
+    pandas.DataFrame.
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:185]:
```suggestion
    Load a table of ship observations of bathymetry off Baja California as a
    pandas.DataFrame.
    
    This is the "@tut_ship.xyz" dataset used in the GMT tutorials.
```

File: pygmt/datasets/samples.py [hunk 3]
```diff
@@ -179,35 +179,17 @@ def load_ocean_ridge_points(**kwargs):
...
 
     Returns
     -------
     data : pandas.DataFrame
         The data table. Columns are longitude, latitude, and bathymetry.
     """
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:191]: I like your change "columns are" to "The column names are" in PR 2308:
```suggestion
    data : pandas.DataFrame
        The data table. The column names are "longitude", "latitude",
        and "bathymetry".
    """
```

File: pygmt/datasets/samples.py [hunk 4]
```diff
@@ -79,6 +78,8 @@ def load_sample_data(name):
 
     # Dictionary of private load functions
     load_func = {
+        "bathymetry": _load_sample_bathymetry,
+        "rock_compositions": _load_rock_sample_compositions,
```
Comment (michaelgrund) [inline pygmt/datasets/samples.py:82]:
```suggestion
```

File: pygmt/datasets/samples.py [hunk 5]
```diff
@@ -79,6 +78,7 @@ def load_sample_data(name):
 
     # Dictionary of private load functions
     load_func = {
+        "bathymetry": _load_sample_bathymetry,
```
Comment (seisman) [inline pygmt/datasets/samples.py:81]: The name `_load_sample_bathymetry` is a little too general. Perhaps we should rename it to something like `_load_baja_california_bathymetry`. 

This function is used internally, so the renaming doesn't cause incompatible changes.

File: pygmt/datasets/samples.py [hunk 6]
```diff
@@ -145,35 +145,18 @@ def _load_ocean_ridge_points():
...
-    if "suppress_warning" not in kwargs:
-        warnings.warn(
-            "This function has been deprecated since v0.6.0 and will be "
-            "removed in v0.9.0. Please use "
-            "load_sample_data(name='bathymetry') instead.",
-            category=FutureWarning,
-            stacklevel=2,
-        )
     fname = which("@tut_ship.xyz", download="c")
     data = pd.read_csv(
```
Comment (seisman) [inline pygmt/datasets/samples.py:161]: Do you mind changing these two lines:
```
    data = pd.read_csv(
        fname, sep="\t", header=None, names=["longitude", "latitude", "bathymetry"]
    )
    return data
```
to 
```
    return pd.read_csv(
        fname, sep="\t", header=None, names=["longitude", "latitude", "bathymetry"]
    )
```

**Pull request merged**